sdk/search/Azure.Search/tests/Serialization/SearchFilterTests.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.Testing;
+using NUnit.Framework;
+
+namespace Azure.Search.Tests
+{
+    public class SearchFilterTests
+    {
+        [Test]
+        public void NoArguments()
+        {
+            Assert.AreEqual("Foo eq 2", SearchFilter.Create($"Foo eq 2"));
+        }
+
+        [Test]
+        public void OneArgument()
+        {
+            Assert.AreEqual("Foo eq 2", SearchFilter.Create($"Foo eq {2}"));
+        }
+
+        [Test]
+        public void ManyArguments()
+        {
+            Assert.AreEqual("Foo eq 2 and Bar eq 3",
```

Comment:
Nit here and below: a purist would say "one assert per unit test". While I think that's overkill, what might be better to consider is using a parameterized test with these inputs and outputs as parameters. When a particular tuple asserts, NUnit (via VS, AzPipelines, etc.) will show a much more helpful message at a glance what failed. I have lots of examples in Key Vault if you want to see, but basically you could do something like this:

```csharp
[TestCaseSource(nameof(ManyArgumentsData))]
public void ManyArguments(FormattableString actual, string expected)
{
    Assert.AreEqual(expected, SearchFilter.Create(actual));
}

private static IEnumerable ManyArgumentsData() => new []
{
    new object[] { $"Foo eq {2} and Bar eq {3}", "Foo eq 2 and Bar eq 3"},
    // ...
}
```

Reply:
Where you can use constants, just use `[TestCase("foo", 1)]` for example.

Reply:
Part of that was not being an expert in the interpolation compiler decision making about when it turns things into constants strings, etc.  But this whole area needs a lot more love and I'd like to clean the tests up when I do that.